vmms/dockerSSH.py

```diff
@@ -0,0 +1,231 @@
+    """
+    p = subprocess.Popen(command, 
+                        stdout=open("/dev/null", 'w'), 
+                        stderr=subprocess.STDOUT)
+    t = 0.0
+    while (t < time_out):
+        ret = p.poll()
+        if ret is None:
+            time.sleep(config.Config.TIMER_POLL_INTERVAL)
+            t += config.Config.TIMER_POLL_INTERVAL
+        elif ret == returnValue:
+            return ret
+        else:
+            p = subprocess.Popen(command,
+                            stdout=open("/dev/null", 'w'),
+                            stderr=subprocess.STDOUT)
+    return ret
+
+def dockerExec(container, cmd, time_out=1):
+    """ docerExec - Executes `docker exec container cmd` and
+        returns output. Container is the name of the docker
+        container and cmd is a list of commands to run.
+    """
+    command = ['docker', 'exec', container, 'sh', '-c'] + cmd
+    return timeout(command, time_out)
+
+#
+# User defined exceptions
+#
+
+class DockerSSH:
+    _OS_X = 'darwin'
+    LOCALHOST = '127.0.0.1'
+
+    def __init__(self):
+        """
+			Checks if the machine is ready to run docker containers.
+            Initialize boot2docker if running on OS X.
+        """
+        try:
+            self.log = logging.getLogger("DockerSSH")
+            # If running on OS X, create a boot2docker VM
+            if sys.platform == self._OS_X:
+                # boot2docker initialization will be part of initial
+                # set up with Tango.
+                self.docker_host_ip = subprocess.check_output(['boot2docker', 'ip']).strip('\n')
+            else:
+                self.docker_host_ip = self.LOCALHOST
+
+            self.log.info("Docker host IP is %s" % self.docker_host_ip)
+
+        except Exception as e:
+            self.log.error(str(e))
+            exit(1)
+
+    def instanceName(self, id, name):
+        """ instanceName - Constructs a VM instance name. Always use
+        this function when you need a VM instance name. Never generate
+        instance names manually.
+        """
+        # return str(id)
+        return "%s-%s-%s" % (config.Config.PREFIX, id, name)
+
+    def domainName(self, vm):
+        """ Returns the domain name that is stored in the vm
+        instance.
+        """
+        return vm.domain_name
+
+    #
+    # VMMS API functions
+    #
+    def initializeVM(self, vm):
+        """ initializeVM -  Nothing to do for initializeVM
+        """
+        return vm
+
+    def waitVM(self, vm, max_secs):
+        """ waitVM - Nothing to do for waitVM
+        """
+        return
+
+    def copyIn(self, vm, inputFiles):
+        """ copyIn - Create a directory to be mounted as a volume
+        for the docker containers. Copy input files to this directory.
+        """
+        instanceName = self.instanceName(vm.id, vm.name)
+
+        # Create a fresh volume
+        volume_path = config.Config.DOCKER_VOLUME_PATH + instanceName +'/'
```

Comment:
Did `DOCKER_VOLUME_PATH` make it into the config.template.py file?

Also, I'm not quite sure what this path is supposed to do (maybe because I can't see an example yet), but would a better name maybe be `DOCKER_VOLUME_ROOT_PATH`?